### PR TITLE
Add tests for `>=`

### LINKED ISSUE #123: `>=`

### PR DESCRIPTION
**Pull request by dgr**
Closes #123

### CODE REVIEW

File: test/clojure/core_test/gt_eq.cljc [hunk 1]
```diff
@@ -0,0 +1,128 @@
+(ns clojure.core-test.gt-eq
+  (:require [clojure.test :as t :refer [are deftest is testing]]
+            [clojure.core-test.portability #?(:cljs :refer-macros :default :refer) [when-var-exists] :as p]))
+
+(when-var-exists >=
+  (deftest test->=
```
Comment (E-A-Griffin) [inline test/clojure/core_test/gt_eq.cljc:6]: nitpick, I think `test>=` is more clear since to me this looks like we're testing some function/macro `->=`

File: test/clojure/core_test/gt_eq.cljc [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        true 0.0M -1.0M
+        true -1 ##-Inf
+        true ##Inf 1
+
+        true 0 0
+        true  1 1
+        true  -1 -1
+        true  ##Inf ##Inf
+        true  ##-Inf ##-Inf
+        false ##NaN ##NaN               ; ##NaN is never equal, even to itself
```
Comment (E-A-Griffin) [inline test/clojure/core_test/gt_eq.cljc:50]: indentation is weird here

File: test/clojure/core_test/gt_eq.cljc [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        true 1 1 0
+        false 0 0 1
+        false 0 1 1
+        true ##Inf 0 ##-Inf
+        false ##-Inf 0 ##Inf)
+      (is (= true (apply >= (reverse (range 10)))))
+      (is (= true (apply >= 10 (reverse (range 10)))))
+      (is (= false (apply >= -1 (reverse (range 10))))))
+
+    (testing "negative tests"
```
Comment (jeaye) [inline test/clojure/core_test/gt_eq.cljc:96]: I think we'll want to scale back on these negative tests, since they're technically undefined behavior in Clojure. The docs string for `>=` says `Returns non-nil if nums are in monotonically non-decreasing order, otherwise false.` so there is an implication that all inputs must be `nums`.

Interestingly, the doc string also says `non-nil`, rather than `true`.

**Review by E-A-Griffin — APPROVED** [review]
Small things but looks good to me!

File: test/clojure/core_test/gt_eq.cljc (hunk 3 above)
Comment (E-A-Griffin) [reply test/clojure/core_test/gt_eq.cljc:96]: Yeah "returns `non-nil`" is very oddly non specific

**Comment by E-A-Griffin** [comment]
One thing that I think could be a good add would be binary numeric comparisons for applicable hosts, Clojure, ClojureScript, ClojureCLR, bb for example have
```
user=> (>= 2r0110 2r0100)
true
```

**Comment by E-A-Griffin** [comment]
Also would be good to test negative Infinity

**Comment by jeaye** [comment]
> One thing that I think could be a good add would be binary numeric comparisons for applicable hosts, Clojure, ClojureScript, ClojureCLR, bb for example have
> 
> ```
> user=> (>= 2r0110 2r0100)
> true
> ```

I like the out-of-the-box thinking, but in this case, `2r0110` is just parsed into a normal number and the behavior of `>=` is not different. We'd just be testing the parser, which is not the goal here.

**Comment by jeaye** [comment]
> Also would be good to test negative Infinity

Good thinking.

File: test/clojure/core_test/gt_eq.cljc (hunk 1 above)
Comment (dgr) [reply test/clojure/core_test/gt_eq.cljc:6]: The template has a standard format: "test-" appended with function name. I'm going to leave this as is unless you have a lot of heartburn, just for consistency with every other test.

File: test/clojure/core_test/gt_eq.cljc (hunk 2 above)
Comment (dgr) [reply test/clojure/core_test/gt_eq.cljc:50]: Fixed.

File: test/clojure/core_test/gt_eq.cljc (hunk 3 above)
Comment (dgr) [reply test/clojure/core_test/gt_eq.cljc:96]: I'll switch the tests to check for non-nil and `false` as opposed to `true` and `false`.

Comment (dgr) [reply test/clojure/core_test/gt_eq.cljc:96]: Also removed all negative tests for anything other than numbers and `nil`.

**Comment by dgr** [comment]
> Also would be good to test negative Infinity

There are multiple tests with `##-Inf`, so I'm not sure what you're looking for here.

**Comment by dgr** [comment]
> > One thing that I think could be a good add would be binary numeric comparisons for applicable hosts, Clojure, ClojureScript, ClojureCLR, bb for example have
> > ```
> > user=> (>= 2r0110 2r0100)
> > true
> > ```
> 
> I like the out-of-the-box thinking, but in this case, `2r0110` is just parsed into a normal number and the behavior of `>=` is not different. We'd just be testing the parser, which is not the goal here.

I agree with this. There's no added benefit as the syntax is just specific to the reader and it's all converted to integers by the time it reaches `>=`.

**Comment by dgr** [comment]
Changing this to draft PR until I can change the tests to check for non-nil rather than `true`.

**Comment by dgr** [comment]
Alright, I think this is ready to go now.

**Comment by jeaye** [comment]
Thanks, Dave!
